src/toolbars/SelectionBar.cpp

```diff
@@ -720,56 +668,23 @@ void SelectionBar::SetSelectionFormat(const NumericFormatSymbol & format)
    }
 }
 
-void SelectionBar::SetRate(double rate)
+void SelectionBar::UpdateRate(double rate)
 {
-   if (rate != mRate) {
+   if (rate != mRate)
+   {
       // if the rate is actually being changed
-      mRate = rate;   // update the stored rate
-      mRateBox->SetValue(wxString::Format(wxT("%d"), (int)rate));
+      mRate = rate; // update the stored rate
 
       // update the TimeTextCtrls if they exist
-      NumericTextCtrl ** Ctrls[5] = { &mStartTime, &mEndTime, &mLengthTime, &mCenterTime, &mAudioTime };
+      NumericTextCtrl** Ctrls[5] = { &mStartTime, &mEndTime, &mLengthTime,
+                                     &mCenterTime, &mAudioTime };
       int i;
-      for(i=0;i<5;i++)
-         if( *Ctrls[i] )
-            (*Ctrls[i])->SetSampleRate( rate );
+      for (i = 0; i < 5; i++)
+         if (*Ctrls[i])
+            (*Ctrls[i])->SetSampleRate(rate);
    }
 }
 
-void SelectionBar::OnRate(wxCommandEvent & WXUNUSED(event))
-{
-   auto value = mRateBox->GetValue();
-
-   if (value.ToDouble(&mRate) && // is a numeric value
-         (mRate != 0.0))
-   {
-      NumericTextCtrl ** Ctrls[5] = { &mStartTime, &mEndTime, &mLengthTime, &mCenterTime, &mAudioTime };
-      int i;
-      for(i=0;i<5;i++)
-         if( *Ctrls[i] )
-            (*Ctrls[i])->SetSampleRate( mRate );
```

Comment:
Does As_SetRate become unused at this commit?

Reply:
Answer, no, but still the function just calls through to `ProjectRate::Get( project ).SetRate( rate );` and there is no need I see for that extra level of abstraction around it in the remaining places.

Reply:
I'll do that cleanup in another branch